.github/workflows/cd.yml

```diff
@@ -6,7 +6,22 @@ on:
       - "v*.*.*"
 
 jobs:
+  check-version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Check version
+        id: check_version
+        run: |
+          VERSION=$(grep '^version =' Cargo.toml | cut -d '"' -f 2)
+          if [[ "$VERSION" != "${{ github.ref }}" ]]; then
+            echo "Version in Cargo.toml ($VERSION) does not match pushed tag (${{ github.ref }})"
```

Comment:
@MordechaiHadad I may be wrong, but it looks like what you're trying to do here is create a job that only succeeds on git tags? So that the later jobs in the `cd` workflow will only run on tags, right?

That makes sense, but I believe Github Actions supports a slightly easier way to do that: tell your `cd` jobs to only run on git tags 
```yaml
on:
  push:
    tags:
      - '*'
```
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags

Reply:
The tag thing is slightly more complicated mainly cuz I want it only to run on "vx.y.z" tags and not any other ones, and I also do the version check (which I have to update slightly) cuz in case I forgot to update the version in the toml file.